### DISCUSSION TITLE
QUIC - Quick UDP Internet Connections in tokio

### DISCUSSION THREAD

**Discussion by hanusek** [tokio-rs/tokio#4992] [Q&A] [answered]
I need to write an async server with encrypted.
Is QUIC support available in tokio.rs?
I cant find information about QUIC in tokio.

**Comment by Noah-Kennedy** [comment, marked as answer]
Tokio does not itself support QUIC, as that is out of scope for the async runtime. That said, there are multiple QUIC implementations you can use _with_ tokio. [quiche](https://docs.rs/quiche/latest/quiche/) is very powerful but somewhat low level, and you may find using it to be a fair bit of work. [quinn](https://docs.rs/quinn/latest/quinn/), on the other hand, is easier to use out of the box as it is higher-level and builds on top of tokio directly for its IO.
